golos.publication/golos.publication.cpp

```diff
@@ -787,4 +791,35 @@ void publication::reblog(name rebloger, name author, std::string permlink) {
             "You can't reblog, because this message doesn't exist.");
 }
 
+int64_t publication::pay_delegators(uint64_t message_id, int64_t claim, name voter, 
+        eosio::symbol tokensymbol, std::vector<structures::delegate_voter> delegate_list) {
+    int64_t dlg_payout_sum = 0;
+    for (auto delegate_obj : delegate_list) {
+        if (delegate_obj.payout_strategy == config::to_delegated_vesting) {
+
+            auto interest_rate = static_cast<uint64_t>(static_cast<uint128_t>(delegate_obj.quantity.amount) * 
+                    delegate_obj.interest_rate / delegate_obj.effective_vesting.amount);
+
+            auto dlg_payout = claim * interest_rate / config::_100percent;
+
+            tables::account_table acc_table(_self, voter.value);
+            auto balance = acc_table.find(tokensymbol.code().raw());
+            acc_table.modify(balance, delegate_obj.delegator, [&](auto& item) {
+                item.delegate_vesting += eosio::asset(dlg_payout, tokensymbol);
+                item.received_vesting += eosio::asset(dlg_payout, tokensymbol);
+            });
+            tables::delegate_table table(_self, tokensymbol.code().raw());
+            auto index_table = table.get_index<"unique"_n>();
+            auto delegate_record = index_table.find(structures::delegate_record::unique_key(delegate_obj.delegator, voter));
+            if (delegate_record != index_table.end()) {
```

Comment:
What happens if the record is missing? Delegate lose this reward? Sum of `delegate_record` must be in consistent with `delegate_vesting` from line 808.